paasta_tools/tron_tools.py

```diff
@@ -138,6 +138,11 @@ def get_requires(self):
     def get_expected_runtime(self):
         return self.config_dict.get('expected_runtime')
 
+    def get_docker_url(self):
+        if self.branch_dict:
+            return super(TronActionConfig, self).get_docker_url()
+        return ''
```

Comment:
Do you need to call get_docker_url in the situation where we don't have deployments?
This pattern looks like it could cause bugs, the other `get_docker_url` will raise if it doesn't have something to work with. Why does tron need to have different behavior?

Reply:
Hm, I've pushed a commit to only call get_docker_url in `format_tron_action_dict` if there's a docker_image (meaning the deployments were loaded). That way, other callers of get_docker_url will get the regular behavior (where it raises an error). Does that look better @solarkennedy?

We still need the empty string fallback in format_tron_action_dict because we use tronfig to validate, which requires generating the full config. For other instance types, we only generate the full config when we're deploying, and we know deployments.json will be there.

Reply:
Hmm. I can't think of any better way to make this cleaner without plumbing a "load_deployments" bool all the way in or something. 

Your explanation for why this is a problem for tron only make sense. shipit.